src/log.ts

```diff
@@ -988,17 +1017,7 @@ class Log implements LogSeverityFunctions {
           Math.max(entry.textPayload.length - delta, 0)
         );
       } else {
-        const fieldsToTruncate = [
-          // Winston:
-          'jsonPayload.fields.metadata.structValue.fields.stack.stringValue',
-          // Bunyan:
-          'jsonPayload.fields.msg.stringValue',
-          'jsonPayload.fields.err.structValue.fields.stack.stringValue',
-          'jsonPayload.fields.err.structValue.fields.message.stringValue',
-          // All:
-          'jsonPayload.fields.message.stringValue',
-        ];
-        for (const field of fieldsToTruncate) {
+        for (const field of this.jsonFieldsToTruncate) {
           const msg: string = dotProp.get(entry, field, '');
           if (msg !== '') {
```

Comment:
nit: i think that `null` will pass this test together with non-empty strings

Reply:
Perhaps I did not understood your comment, but `this.jsonFieldsToTruncate` should not be null... It can be directly accessed and set to null, but I guess this is not something we should handle.
I added a check for null strings when we update `this.jsonFieldsToTruncate`, let me know if thats what you mean to cover.

Reply:
Sorry for misleading you. I meant the line 1022: `if (msg !== '') {`

Reply:
Sure, fixing